source/request_response.c

```diff
@@ -630,15 +630,24 @@ int aws_http_message_get_request_method(
     return aws_raise_error(error);
 }
 
+static struct aws_byte_cursor s_default_empty_path = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("/");
+
 int aws_http_message_set_request_path(struct aws_http_message *request_message, struct aws_byte_cursor path) {
     AWS_PRECONDITION(request_message);
     AWS_PRECONDITION(aws_byte_cursor_is_valid(&path));
     AWS_PRECONDITION(request_message->request_data);
 
+    /* Its valid for uri path to be empty, but http spec requires empty paths to
+    be sent as "/", so default it here. */
+    if (path.len == 0) {
+        path = s_default_empty_path;
+    }
+
     if (request_message->request_data) {
         switch (request_message->http_version) {
-            case AWS_HTTP_VERSION_1_1:
+            case AWS_HTTP_VERSION_1_1: {
                 return s_set_string_from_cursor(&request_message->request_data->path, path, request_message->allocator);
+            }
             case AWS_HTTP_VERSION_2:
                 return aws_http2_headers_set_request_path(request_message->headers, path);
```

Comment:
Let's also update for `aws_http2_headers_set_request_path`, when the path cursor is empty, use the default one

Reply:
i missed that aws_http2_headers_set_request_path was public api. added logic to both paths